// @leonardo-ai/sdk/sdk/models/shared/custommodeltype.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomModelType$ = exports.CustomModelType$outboundSchema = exports.CustomModelType$inboundSchema = exports.CustomModelType = void 0;
const z = __importStar(require("zod"));
/**
 * The category the most accurately reflects the model.
 */
var CustomModelType;
(function (CustomModelType) {
    CustomModelType["General"] = "GENERAL";
    CustomModelType["Buildings"] = "BUILDINGS";
    CustomModelType["Characters"] = "CHARACTERS";
    CustomModelType["Environments"] = "ENVIRONMENTS";
    CustomModelType["Fashion"] = "FASHION";
    CustomModelType["Illustrations"] = "ILLUSTRATIONS";
    CustomModelType["GameItems"] = "GAME_ITEMS";
    CustomModelType["GraphicalElements"] = "GRAPHICAL_ELEMENTS";
    CustomModelType["Photography"] = "PHOTOGRAPHY";
    CustomModelType["PixelArt"] = "PIXEL_ART";
    CustomModelType["ProductDesign"] = "PRODUCT_DESIGN";
    CustomModelType["Textures"] = "TEXTURES";
    CustomModelType["UiElements"] = "UI_ELEMENTS";
    CustomModelType["Vector"] = "VECTOR";
})(CustomModelType || (exports.CustomModelType = CustomModelType = {}));
/** @internal */
exports.CustomModelType$inboundSchema = z.nativeEnum(CustomModelType);
/** @internal */
exports.CustomModelType$outboundSchema = exports.CustomModelType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomModelType$;
(function (CustomModelType$) {
    /** @deprecated use `CustomModelType$inboundSchema` instead. */
    CustomModelType$.inboundSchema = exports.CustomModelType$inboundSchema;
    /** @deprecated use `CustomModelType$outboundSchema` instead. */
    CustomModelType$.outboundSchema = exports.CustomModelType$outboundSchema;
})(CustomModelType$ || (exports.CustomModelType$ = CustomModelType$ = {}));
