// @leonardo-ai/sdk/sdk/models/shared/elementinput.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.elementInputFromJSON = exports.elementInputToJSON = exports.ElementInput$ = exports.ElementInput$outboundSchema = exports.ElementInput$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.ElementInput$inboundSchema = z.object({
    akUUID: z.nullable(z.string()).optional(),
    weight: z.nullable(z.number()).optional(),
});
/** @internal */
exports.ElementInput$outboundSchema = z.object({
    akUUID: z.nullable(z.string()).optional(),
    weight: z.nullable(z.number()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ElementInput$;
(function (ElementInput$) {
    /** @deprecated use `ElementInput$inboundSchema` instead. */
    ElementInput$.inboundSchema = exports.ElementInput$inboundSchema;
    /** @deprecated use `ElementInput$outboundSchema` instead. */
    ElementInput$.outboundSchema = exports.ElementInput$outboundSchema;
})(ElementInput$ || (exports.ElementInput$ = ElementInput$ = {}));
function elementInputToJSON(elementInput) {
    return JSON.stringify(exports.ElementInput$outboundSchema.parse(elementInput));
}
exports.elementInputToJSON = elementInputToJSON;
function elementInputFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ElementInput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ElementInput' from JSON`);
}
exports.elementInputFromJSON = elementInputFromJSON;
