// @leonardo-ai/sdk/sdk/models/shared/lcmgenerationstyle.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LcmGenerationStyle$ = exports.LcmGenerationStyle$outboundSchema = exports.LcmGenerationStyle$inboundSchema = exports.LcmGenerationStyle = void 0;
const z = __importStar(require("zod"));
/**
 * The style to generate LCM images with.
 */
var LcmGenerationStyle;
(function (LcmGenerationStyle) {
    LcmGenerationStyle["Anime"] = "ANIME";
    LcmGenerationStyle["Cinematic"] = "CINEMATIC";
    LcmGenerationStyle["DigitalArt"] = "DIGITAL_ART";
    LcmGenerationStyle["Dynamic"] = "DYNAMIC";
    LcmGenerationStyle["Environment"] = "ENVIRONMENT";
    LcmGenerationStyle["FantasyArt"] = "FANTASY_ART";
    LcmGenerationStyle["Illustration"] = "ILLUSTRATION";
    LcmGenerationStyle["Photography"] = "PHOTOGRAPHY";
    LcmGenerationStyle["Render3D"] = "RENDER_3D";
    LcmGenerationStyle["Raytraced"] = "RAYTRACED";
    LcmGenerationStyle["SketchBw"] = "SKETCH_BW";
    LcmGenerationStyle["SketchColor"] = "SKETCH_COLOR";
    LcmGenerationStyle["Vibrant"] = "VIBRANT";
    LcmGenerationStyle["None"] = "NONE";
})(LcmGenerationStyle || (exports.LcmGenerationStyle = LcmGenerationStyle = {}));
/** @internal */
exports.LcmGenerationStyle$inboundSchema = z.nativeEnum(LcmGenerationStyle);
/** @internal */
exports.LcmGenerationStyle$outboundSchema = exports.LcmGenerationStyle$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LcmGenerationStyle$;
(function (LcmGenerationStyle$) {
    /** @deprecated use `LcmGenerationStyle$inboundSchema` instead. */
    LcmGenerationStyle$.inboundSchema = exports.LcmGenerationStyle$inboundSchema;
    /** @deprecated use `LcmGenerationStyle$outboundSchema` instead. */
    LcmGenerationStyle$.outboundSchema = exports.LcmGenerationStyle$outboundSchema;
})(LcmGenerationStyle$ || (exports.LcmGenerationStyle$ = LcmGenerationStyle$ = {}));
