// @leonardo-ai/sdk/sdk/models/shared/pricingcalculatorservices.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricingCalculatorServices$ = exports.PricingCalculatorServices$outboundSchema = exports.PricingCalculatorServices$inboundSchema = exports.PricingCalculatorServices = void 0;
const z = __importStar(require("zod"));
/**
 * The services to be chosen for calculating the API credit cost.
 */
var PricingCalculatorServices;
(function (PricingCalculatorServices) {
    PricingCalculatorServices["ImageGeneration"] = "IMAGE_GENERATION";
    PricingCalculatorServices["FantasyAvatarGeneration"] = "FANTASY_AVATAR_GENERATION";
    PricingCalculatorServices["MotionGeneration"] = "MOTION_GENERATION";
    PricingCalculatorServices["LcmGeneration"] = "LCM_GENERATION";
    PricingCalculatorServices["ModelTraining"] = "MODEL_TRAINING";
    PricingCalculatorServices["TextureGeneration"] = "TEXTURE_GENERATION";
    PricingCalculatorServices["UniversalUpscaler"] = "UNIVERSAL_UPSCALER";
    PricingCalculatorServices["UniversalUpscalerUltra"] = "UNIVERSAL_UPSCALER_ULTRA";
})(PricingCalculatorServices || (exports.PricingCalculatorServices = PricingCalculatorServices = {}));
/** @internal */
exports.PricingCalculatorServices$inboundSchema = z.nativeEnum(PricingCalculatorServices);
/** @internal */
exports.PricingCalculatorServices$outboundSchema = exports.PricingCalculatorServices$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorServices$;
(function (PricingCalculatorServices$) {
    /** @deprecated use `PricingCalculatorServices$inboundSchema` instead. */
    PricingCalculatorServices$.inboundSchema = exports.PricingCalculatorServices$inboundSchema;
    /** @deprecated use `PricingCalculatorServices$outboundSchema` instead. */
    PricingCalculatorServices$.outboundSchema = exports.PricingCalculatorServices$outboundSchema;
})(PricingCalculatorServices$ || (exports.PricingCalculatorServices$ = PricingCalculatorServices$ = {}));
