// @leonardo-ai/sdk/sdk/models/shared/sdgenerationstyle.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SdGenerationStyle$ = exports.SdGenerationStyle$outboundSchema = exports.SdGenerationStyle$inboundSchema = exports.SdGenerationStyle = void 0;
const z = __importStar(require("zod"));
/**
 * The style to generate images with. When photoReal is enabled, refer to the Guide section for a full list. When alchemy is disabled, use LEONARDO or NONE. When alchemy is enabled, use ANIME, CREATIVE, DYNAMIC, ENVIRONMENT, GENERAL, ILLUSTRATION, PHOTOGRAPHY, RAYTRACED, RENDER_3D, SKETCH_BW, SKETCH_COLOR, or NONE.
 */
var SdGenerationStyle;
(function (SdGenerationStyle) {
    SdGenerationStyle["Anime"] = "ANIME";
    SdGenerationStyle["Bokeh"] = "BOKEH";
    SdGenerationStyle["Cinematic"] = "CINEMATIC";
    SdGenerationStyle["CinematicCloseup"] = "CINEMATIC_CLOSEUP";
    SdGenerationStyle["Creative"] = "CREATIVE";
    SdGenerationStyle["Dynamic"] = "DYNAMIC";
    SdGenerationStyle["Environment"] = "ENVIRONMENT";
    SdGenerationStyle["Fashion"] = "FASHION";
    SdGenerationStyle["Film"] = "FILM";
    SdGenerationStyle["Food"] = "FOOD";
    SdGenerationStyle["General"] = "GENERAL";
    SdGenerationStyle["Hdr"] = "HDR";
    SdGenerationStyle["Illustration"] = "ILLUSTRATION";
    SdGenerationStyle["Leonardo"] = "LEONARDO";
    SdGenerationStyle["LongExposure"] = "LONG_EXPOSURE";
    SdGenerationStyle["Macro"] = "MACRO";
    SdGenerationStyle["Minimalistic"] = "MINIMALISTIC";
    SdGenerationStyle["Monochrome"] = "MONOCHROME";
    SdGenerationStyle["Moody"] = "MOODY";
    SdGenerationStyle["None"] = "NONE";
    SdGenerationStyle["Neutral"] = "NEUTRAL";
    SdGenerationStyle["Photography"] = "PHOTOGRAPHY";
    SdGenerationStyle["Portrait"] = "PORTRAIT";
    SdGenerationStyle["Raytraced"] = "RAYTRACED";
    SdGenerationStyle["Render3D"] = "RENDER_3D";
    SdGenerationStyle["Retro"] = "RETRO";
    SdGenerationStyle["SketchBw"] = "SKETCH_BW";
    SdGenerationStyle["SketchColor"] = "SKETCH_COLOR";
    SdGenerationStyle["StockPhoto"] = "STOCK_PHOTO";
    SdGenerationStyle["Vibrant"] = "VIBRANT";
    SdGenerationStyle["Unprocessed"] = "UNPROCESSED";
})(SdGenerationStyle || (exports.SdGenerationStyle = SdGenerationStyle = {}));
/** @internal */
exports.SdGenerationStyle$inboundSchema = z.nativeEnum(SdGenerationStyle);
/** @internal */
exports.SdGenerationStyle$outboundSchema = exports.SdGenerationStyle$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SdGenerationStyle$;
(function (SdGenerationStyle$) {
    /** @deprecated use `SdGenerationStyle$inboundSchema` instead. */
    SdGenerationStyle$.inboundSchema = exports.SdGenerationStyle$inboundSchema;
    /** @deprecated use `SdGenerationStyle$outboundSchema` instead. */
    SdGenerationStyle$.outboundSchema = exports.SdGenerationStyle$outboundSchema;
})(SdGenerationStyle$ || (exports.SdGenerationStyle$ = SdGenerationStyle$ = {}));
