// @leonardo-ai/sdk/sdk/models/shared/sdgenerationschedulers.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SdGenerationSchedulers$ = exports.SdGenerationSchedulers$outboundSchema = exports.SdGenerationSchedulers$inboundSchema = exports.SdGenerationSchedulers = void 0;
const z = __importStar(require("zod"));
/**
 * The scheduler to generate images with. Defaults to EULER_DISCRETE if not specified.
 */
var SdGenerationSchedulers;
(function (SdGenerationSchedulers) {
    SdGenerationSchedulers["Klms"] = "KLMS";
    SdGenerationSchedulers["EulerAncestralDiscrete"] = "EULER_ANCESTRAL_DISCRETE";
    SdGenerationSchedulers["EulerDiscrete"] = "EULER_DISCRETE";
    SdGenerationSchedulers["Ddim"] = "DDIM";
    SdGenerationSchedulers["DpmSolver"] = "DPM_SOLVER";
    SdGenerationSchedulers["Pndm"] = "PNDM";
    SdGenerationSchedulers["Leonardo"] = "LEONARDO";
})(SdGenerationSchedulers || (exports.SdGenerationSchedulers = SdGenerationSchedulers = {}));
/** @internal */
exports.SdGenerationSchedulers$inboundSchema = z.nativeEnum(SdGenerationSchedulers);
/** @internal */
exports.SdGenerationSchedulers$outboundSchema = exports.SdGenerationSchedulers$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SdGenerationSchedulers$;
(function (SdGenerationSchedulers$) {
    /** @deprecated use `SdGenerationSchedulers$inboundSchema` instead. */
    SdGenerationSchedulers$.inboundSchema = exports.SdGenerationSchedulers$inboundSchema;
    /** @deprecated use `SdGenerationSchedulers$outboundSchema` instead. */
    SdGenerationSchedulers$.outboundSchema = exports.SdGenerationSchedulers$outboundSchema;
})(SdGenerationSchedulers$ || (exports.SdGenerationSchedulers$ = SdGenerationSchedulers$ = {}));
