// @leonardo-ai/sdk/sdk/models/shared/strength.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Strength$ = exports.Strength$outboundSchema = exports.Strength$inboundSchema = exports.Strength = void 0;
const z = __importStar(require("zod"));
/**
 * When training using the PIXEL_ART model type, this influences the training strength.
 */
var Strength;
(function (Strength) {
    Strength["VeryLow"] = "VERY_LOW";
    Strength["Low"] = "LOW";
    Strength["Medium"] = "MEDIUM";
    Strength["High"] = "HIGH";
})(Strength || (exports.Strength = Strength = {}));
/** @internal */
exports.Strength$inboundSchema = z
    .nativeEnum(Strength);
/** @internal */
exports.Strength$outboundSchema = exports.Strength$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Strength$;
(function (Strength$) {
    /** @deprecated use `Strength$inboundSchema` instead. */
    Strength$.inboundSchema = exports.Strength$inboundSchema;
    /** @deprecated use `Strength$outboundSchema` instead. */
    Strength$.outboundSchema = exports.Strength$outboundSchema;
})(Strength$ || (exports.Strength$ = Strength$ = {}));
