// @leonardo-ai/sdk/sdk/models/shared/universalupscalerultrastyle.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UniversalUpscalerUltraStyle$ = exports.UniversalUpscalerUltraStyle$outboundSchema = exports.UniversalUpscalerUltraStyle$inboundSchema = exports.UniversalUpscalerUltraStyle = void 0;
const z = __importStar(require("zod"));
/**
 * The ultra style to upscale images using universal upscaler with. Can not be used with upscalerStyle.
 */
var UniversalUpscalerUltraStyle;
(function (UniversalUpscalerUltraStyle) {
    UniversalUpscalerUltraStyle["Artistic"] = "ARTISTIC";
    UniversalUpscalerUltraStyle["Realistic"] = "REALISTIC";
})(UniversalUpscalerUltraStyle || (exports.UniversalUpscalerUltraStyle = UniversalUpscalerUltraStyle = {}));
/** @internal */
exports.UniversalUpscalerUltraStyle$inboundSchema = z.nativeEnum(UniversalUpscalerUltraStyle);
/** @internal */
exports.UniversalUpscalerUltraStyle$outboundSchema = exports.UniversalUpscalerUltraStyle$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UniversalUpscalerUltraStyle$;
(function (UniversalUpscalerUltraStyle$) {
    /** @deprecated use `UniversalUpscalerUltraStyle$inboundSchema` instead. */
    UniversalUpscalerUltraStyle$.inboundSchema = exports.UniversalUpscalerUltraStyle$inboundSchema;
    /** @deprecated use `UniversalUpscalerUltraStyle$outboundSchema` instead. */
    UniversalUpscalerUltraStyle$.outboundSchema = exports.UniversalUpscalerUltraStyle$outboundSchema;
})(UniversalUpscalerUltraStyle$ || (exports.UniversalUpscalerUltraStyle$ = UniversalUpscalerUltraStyle$ = {}));
