// @leonardo-ai/sdk/sdk/models/shared/userelementsinput.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.userElementsInputFromJSON = exports.userElementsInputToJSON = exports.UserElementsInput$ = exports.UserElementsInput$outboundSchema = exports.UserElementsInput$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.UserElementsInput$inboundSchema = z.object({
    userLoraId: z.nullable(z.number()).optional(),
    weight: z.nullable(z.number()).optional(),
});
/** @internal */
exports.UserElementsInput$outboundSchema = z.object({
    userLoraId: z.nullable(z.number()).optional(),
    weight: z.nullable(z.number()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserElementsInput$;
(function (UserElementsInput$) {
    /** @deprecated use `UserElementsInput$inboundSchema` instead. */
    UserElementsInput$.inboundSchema = exports.UserElementsInput$inboundSchema;
    /** @deprecated use `UserElementsInput$outboundSchema` instead. */
    UserElementsInput$.outboundSchema = exports.UserElementsInput$outboundSchema;
})(UserElementsInput$ || (exports.UserElementsInput$ = UserElementsInput$ = {}));
function userElementsInputToJSON(userElementsInput) {
    return JSON.stringify(exports.UserElementsInput$outboundSchema.parse(userElementsInput));
}
exports.userElementsInputToJSON = userElementsInputToJSON;
function userElementsInputFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UserElementsInput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UserElementsInput' from JSON`);
}
exports.userElementsInputFromJSON = userElementsInputFromJSON;
