// @leonardo-ai/sdk/sdk/variation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Variation = void 0;
const variationCreateUniversalUpscalerJob_js_1 = require("../funcs/variationCreateUniversalUpscalerJob.js");
const variationCreateVariationNoBG_js_1 = require("../funcs/variationCreateVariationNoBG.js");
const variationCreateVariationUnzoom_js_1 = require("../funcs/variationCreateVariationUnzoom.js");
const variationCreateVariationUpscale_js_1 = require("../funcs/variationCreateVariationUpscale.js");
const variationGetVariationById_js_1 = require("../funcs/variationGetVariationById.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("./types/fp.js");
class Variation extends sdks_js_1.ClientSDK {
    /**
     * Create using Universal Upscaler
     *
     * @remarks
     * This endpoint will create a high resolution image using Universal Upscaler
     */
    async createUniversalUpscalerJob(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, variationCreateUniversalUpscalerJob_js_1.variationCreateUniversalUpscalerJob)(this, request, options));
    }
    /**
     * Create no background
     *
     * @remarks
     * This endpoint will create a no background variation of the provided image ID
     */
    async createVariationNoBG(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, variationCreateVariationNoBG_js_1.variationCreateVariationNoBG)(this, request, options));
    }
    /**
     * Create unzoom
     *
     * @remarks
     * This endpoint will create an unzoom variation for the provided image ID
     */
    async createVariationUnzoom(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, variationCreateVariationUnzoom_js_1.variationCreateVariationUnzoom)(this, request, options));
    }
    /**
     * Create upscale
     *
     * @remarks
     * This endpoint will create an upscale for the provided image ID
     */
    async createVariationUpscale(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, variationCreateVariationUpscale_js_1.variationCreateVariationUpscale)(this, request, options));
    }
    /**
     * Get variation by ID
     *
     * @remarks
     * This endpoint will get the variation by ID
     */
    async getVariationById(id, options) {
        return (0, fp_js_1.unwrapAsync)((0, variationGetVariationById_js_1.variationGetVariationById)(this, id, options));
    }
}
exports.Variation = Variation;
