// @leonardo-ai/sdk/sdk/models/operations/createvariationnobg.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createVariationNoBGResponseFromJSON = exports.createVariationNoBGResponseToJSON = exports.CreateVariationNoBGResponse$ = exports.CreateVariationNoBGResponse$outboundSchema = exports.CreateVariationNoBGResponse$inboundSchema = exports.createVariationNoBGResponseBodyFromJSON = exports.createVariationNoBGResponseBodyToJSON = exports.CreateVariationNoBGResponseBody$ = exports.CreateVariationNoBGResponseBody$outboundSchema = exports.CreateVariationNoBGResponseBody$inboundSchema = exports.sdUpscaleJobOutputFromJSON = exports.sdUpscaleJobOutputToJSON = exports.SDUpscaleJobOutput$ = exports.SDUpscaleJobOutput$outboundSchema = exports.SDUpscaleJobOutput$inboundSchema = exports.createVariationNoBGRequestBodyFromJSON = exports.createVariationNoBGRequestBodyToJSON = exports.CreateVariationNoBGRequestBody$ = exports.CreateVariationNoBGRequestBody$outboundSchema = exports.CreateVariationNoBGRequestBody$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.CreateVariationNoBGRequestBody$inboundSchema = z.object({
    id: z.string(),
    isVariation: z.boolean().optional(),
});
/** @internal */
exports.CreateVariationNoBGRequestBody$outboundSchema = z.object({
    id: z.string(),
    isVariation: z.boolean().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateVariationNoBGRequestBody$;
(function (CreateVariationNoBGRequestBody$) {
    /** @deprecated use `CreateVariationNoBGRequestBody$inboundSchema` instead. */
    CreateVariationNoBGRequestBody$.inboundSchema = exports.CreateVariationNoBGRequestBody$inboundSchema;
    /** @deprecated use `CreateVariationNoBGRequestBody$outboundSchema` instead. */
    CreateVariationNoBGRequestBody$.outboundSchema = exports.CreateVariationNoBGRequestBody$outboundSchema;
})(CreateVariationNoBGRequestBody$ || (exports.CreateVariationNoBGRequestBody$ = CreateVariationNoBGRequestBody$ = {}));
function createVariationNoBGRequestBodyToJSON(createVariationNoBGRequestBody) {
    return JSON.stringify(exports.CreateVariationNoBGRequestBody$outboundSchema.parse(createVariationNoBGRequestBody));
}
exports.createVariationNoBGRequestBodyToJSON = createVariationNoBGRequestBodyToJSON;
function createVariationNoBGRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateVariationNoBGRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateVariationNoBGRequestBody' from JSON`);
}
exports.createVariationNoBGRequestBodyFromJSON = createVariationNoBGRequestBodyFromJSON;
/** @internal */
exports.SDUpscaleJobOutput$inboundSchema = z.object({
    apiCreditCost: z.nullable(z.number().int()).optional(),
    id: z.nullable(z.string()).optional(),
});
/** @internal */
exports.SDUpscaleJobOutput$outboundSchema = z.object({
    apiCreditCost: z.nullable(z.number().int()).optional(),
    id: z.nullable(z.string()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SDUpscaleJobOutput$;
(function (SDUpscaleJobOutput$) {
    /** @deprecated use `SDUpscaleJobOutput$inboundSchema` instead. */
    SDUpscaleJobOutput$.inboundSchema = exports.SDUpscaleJobOutput$inboundSchema;
    /** @deprecated use `SDUpscaleJobOutput$outboundSchema` instead. */
    SDUpscaleJobOutput$.outboundSchema = exports.SDUpscaleJobOutput$outboundSchema;
})(SDUpscaleJobOutput$ || (exports.SDUpscaleJobOutput$ = SDUpscaleJobOutput$ = {}));
function sdUpscaleJobOutputToJSON(sdUpscaleJobOutput) {
    return JSON.stringify(exports.SDUpscaleJobOutput$outboundSchema.parse(sdUpscaleJobOutput));
}
exports.sdUpscaleJobOutputToJSON = sdUpscaleJobOutputToJSON;
function sdUpscaleJobOutputFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SDUpscaleJobOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SDUpscaleJobOutput' from JSON`);
}
exports.sdUpscaleJobOutputFromJSON = sdUpscaleJobOutputFromJSON;
/** @internal */
exports.CreateVariationNoBGResponseBody$inboundSchema = z.object({
    sdNobgJob: z.lazy(() => exports.SDUpscaleJobOutput$inboundSchema).optional(),
});
/** @internal */
exports.CreateVariationNoBGResponseBody$outboundSchema = z.object({
    sdNobgJob: z.lazy(() => exports.SDUpscaleJobOutput$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateVariationNoBGResponseBody$;
(function (CreateVariationNoBGResponseBody$) {
    /** @deprecated use `CreateVariationNoBGResponseBody$inboundSchema` instead. */
    CreateVariationNoBGResponseBody$.inboundSchema = exports.CreateVariationNoBGResponseBody$inboundSchema;
    /** @deprecated use `CreateVariationNoBGResponseBody$outboundSchema` instead. */
    CreateVariationNoBGResponseBody$.outboundSchema = exports.CreateVariationNoBGResponseBody$outboundSchema;
})(CreateVariationNoBGResponseBody$ || (exports.CreateVariationNoBGResponseBody$ = CreateVariationNoBGResponseBody$ = {}));
function createVariationNoBGResponseBodyToJSON(createVariationNoBGResponseBody) {
    return JSON.stringify(exports.CreateVariationNoBGResponseBody$outboundSchema.parse(createVariationNoBGResponseBody));
}
exports.createVariationNoBGResponseBodyToJSON = createVariationNoBGResponseBodyToJSON;
function createVariationNoBGResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateVariationNoBGResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateVariationNoBGResponseBody' from JSON`);
}
exports.createVariationNoBGResponseBodyFromJSON = createVariationNoBGResponseBodyFromJSON;
/** @internal */
exports.CreateVariationNoBGResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    object: z.lazy(() => exports.CreateVariationNoBGResponseBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.CreateVariationNoBGResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    object: z.lazy(() => exports.CreateVariationNoBGResponseBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateVariationNoBGResponse$;
(function (CreateVariationNoBGResponse$) {
    /** @deprecated use `CreateVariationNoBGResponse$inboundSchema` instead. */
    CreateVariationNoBGResponse$.inboundSchema = exports.CreateVariationNoBGResponse$inboundSchema;
    /** @deprecated use `CreateVariationNoBGResponse$outboundSchema` instead. */
    CreateVariationNoBGResponse$.outboundSchema = exports.CreateVariationNoBGResponse$outboundSchema;
})(CreateVariationNoBGResponse$ || (exports.CreateVariationNoBGResponse$ = CreateVariationNoBGResponse$ = {}));
function createVariationNoBGResponseToJSON(createVariationNoBGResponse) {
    return JSON.stringify(exports.CreateVariationNoBGResponse$outboundSchema.parse(createVariationNoBGResponse));
}
exports.createVariationNoBGResponseToJSON = createVariationNoBGResponseToJSON;
function createVariationNoBGResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateVariationNoBGResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateVariationNoBGResponse' from JSON`);
}
exports.createVariationNoBGResponseFromJSON = createVariationNoBGResponseFromJSON;
