// @leonardo-ai/sdk/sdk/models/operations/createvariationupscale.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createVariationUpscaleResponseFromJSON = exports.createVariationUpscaleResponseToJSON = exports.CreateVariationUpscaleResponse$ = exports.CreateVariationUpscaleResponse$outboundSchema = exports.CreateVariationUpscaleResponse$inboundSchema = exports.createVariationUpscaleResponseBodyFromJSON = exports.createVariationUpscaleResponseBodyToJSON = exports.CreateVariationUpscaleResponseBody$ = exports.CreateVariationUpscaleResponseBody$outboundSchema = exports.CreateVariationUpscaleResponseBody$inboundSchema = exports.createVariationUpscaleSDUpscaleJobOutputFromJSON = exports.createVariationUpscaleSDUpscaleJobOutputToJSON = exports.CreateVariationUpscaleSDUpscaleJobOutput$ = exports.CreateVariationUpscaleSDUpscaleJobOutput$outboundSchema = exports.CreateVariationUpscaleSDUpscaleJobOutput$inboundSchema = exports.createVariationUpscaleRequestBodyFromJSON = exports.createVariationUpscaleRequestBodyToJSON = exports.CreateVariationUpscaleRequestBody$ = exports.CreateVariationUpscaleRequestBody$outboundSchema = exports.CreateVariationUpscaleRequestBody$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.CreateVariationUpscaleRequestBody$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.CreateVariationUpscaleRequestBody$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateVariationUpscaleRequestBody$;
(function (CreateVariationUpscaleRequestBody$) {
    /** @deprecated use `CreateVariationUpscaleRequestBody$inboundSchema` instead. */
    CreateVariationUpscaleRequestBody$.inboundSchema = exports.CreateVariationUpscaleRequestBody$inboundSchema;
    /** @deprecated use `CreateVariationUpscaleRequestBody$outboundSchema` instead. */
    CreateVariationUpscaleRequestBody$.outboundSchema = exports.CreateVariationUpscaleRequestBody$outboundSchema;
})(CreateVariationUpscaleRequestBody$ || (exports.CreateVariationUpscaleRequestBody$ = CreateVariationUpscaleRequestBody$ = {}));
function createVariationUpscaleRequestBodyToJSON(createVariationUpscaleRequestBody) {
    return JSON.stringify(exports.CreateVariationUpscaleRequestBody$outboundSchema.parse(createVariationUpscaleRequestBody));
}
exports.createVariationUpscaleRequestBodyToJSON = createVariationUpscaleRequestBodyToJSON;
function createVariationUpscaleRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateVariationUpscaleRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateVariationUpscaleRequestBody' from JSON`);
}
exports.createVariationUpscaleRequestBodyFromJSON = createVariationUpscaleRequestBodyFromJSON;
/** @internal */
exports.CreateVariationUpscaleSDUpscaleJobOutput$inboundSchema = z.object({
    apiCreditCost: z.nullable(z.number().int()).optional(),
    id: z.string().optional(),
});
/** @internal */
exports.CreateVariationUpscaleSDUpscaleJobOutput$outboundSchema = z.object({
    apiCreditCost: z.nullable(z.number().int()).optional(),
    id: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateVariationUpscaleSDUpscaleJobOutput$;
(function (CreateVariationUpscaleSDUpscaleJobOutput$) {
    /** @deprecated use `CreateVariationUpscaleSDUpscaleJobOutput$inboundSchema` instead. */
    CreateVariationUpscaleSDUpscaleJobOutput$.inboundSchema = exports.CreateVariationUpscaleSDUpscaleJobOutput$inboundSchema;
    /** @deprecated use `CreateVariationUpscaleSDUpscaleJobOutput$outboundSchema` instead. */
    CreateVariationUpscaleSDUpscaleJobOutput$.outboundSchema = exports.CreateVariationUpscaleSDUpscaleJobOutput$outboundSchema;
})(CreateVariationUpscaleSDUpscaleJobOutput$ || (exports.CreateVariationUpscaleSDUpscaleJobOutput$ = CreateVariationUpscaleSDUpscaleJobOutput$ = {}));
function createVariationUpscaleSDUpscaleJobOutputToJSON(createVariationUpscaleSDUpscaleJobOutput) {
    return JSON.stringify(exports.CreateVariationUpscaleSDUpscaleJobOutput$outboundSchema.parse(createVariationUpscaleSDUpscaleJobOutput));
}
exports.createVariationUpscaleSDUpscaleJobOutputToJSON = createVariationUpscaleSDUpscaleJobOutputToJSON;
function createVariationUpscaleSDUpscaleJobOutputFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateVariationUpscaleSDUpscaleJobOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateVariationUpscaleSDUpscaleJobOutput' from JSON`);
}
exports.createVariationUpscaleSDUpscaleJobOutputFromJSON = createVariationUpscaleSDUpscaleJobOutputFromJSON;
/** @internal */
exports.CreateVariationUpscaleResponseBody$inboundSchema = z.object({
    sdUpscaleJob: z.nullable(z.lazy(() => exports.CreateVariationUpscaleSDUpscaleJobOutput$inboundSchema)).optional(),
});
/** @internal */
exports.CreateVariationUpscaleResponseBody$outboundSchema = z.object({
    sdUpscaleJob: z.nullable(z.lazy(() => exports.CreateVariationUpscaleSDUpscaleJobOutput$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateVariationUpscaleResponseBody$;
(function (CreateVariationUpscaleResponseBody$) {
    /** @deprecated use `CreateVariationUpscaleResponseBody$inboundSchema` instead. */
    CreateVariationUpscaleResponseBody$.inboundSchema = exports.CreateVariationUpscaleResponseBody$inboundSchema;
    /** @deprecated use `CreateVariationUpscaleResponseBody$outboundSchema` instead. */
    CreateVariationUpscaleResponseBody$.outboundSchema = exports.CreateVariationUpscaleResponseBody$outboundSchema;
})(CreateVariationUpscaleResponseBody$ || (exports.CreateVariationUpscaleResponseBody$ = CreateVariationUpscaleResponseBody$ = {}));
function createVariationUpscaleResponseBodyToJSON(createVariationUpscaleResponseBody) {
    return JSON.stringify(exports.CreateVariationUpscaleResponseBody$outboundSchema.parse(createVariationUpscaleResponseBody));
}
exports.createVariationUpscaleResponseBodyToJSON = createVariationUpscaleResponseBodyToJSON;
function createVariationUpscaleResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateVariationUpscaleResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateVariationUpscaleResponseBody' from JSON`);
}
exports.createVariationUpscaleResponseBodyFromJSON = createVariationUpscaleResponseBodyFromJSON;
/** @internal */
exports.CreateVariationUpscaleResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    object: z.lazy(() => exports.CreateVariationUpscaleResponseBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.CreateVariationUpscaleResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    object: z.lazy(() => exports.CreateVariationUpscaleResponseBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateVariationUpscaleResponse$;
(function (CreateVariationUpscaleResponse$) {
    /** @deprecated use `CreateVariationUpscaleResponse$inboundSchema` instead. */
    CreateVariationUpscaleResponse$.inboundSchema = exports.CreateVariationUpscaleResponse$inboundSchema;
    /** @deprecated use `CreateVariationUpscaleResponse$outboundSchema` instead. */
    CreateVariationUpscaleResponse$.outboundSchema = exports.CreateVariationUpscaleResponse$outboundSchema;
})(CreateVariationUpscaleResponse$ || (exports.CreateVariationUpscaleResponse$ = CreateVariationUpscaleResponse$ = {}));
function createVariationUpscaleResponseToJSON(createVariationUpscaleResponse) {
    return JSON.stringify(exports.CreateVariationUpscaleResponse$outboundSchema.parse(createVariationUpscaleResponse));
}
exports.createVariationUpscaleResponseToJSON = createVariationUpscaleResponseToJSON;
function createVariationUpscaleResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateVariationUpscaleResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateVariationUpscaleResponse' from JSON`);
}
exports.createVariationUpscaleResponseFromJSON = createVariationUpscaleResponseFromJSON;
