// @leonardo-ai/sdk/sdk/models/operations/createsvdmotiongeneration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createSVDMotionGenerationResponseFromJSON = exports.createSVDMotionGenerationResponseToJSON = exports.CreateSVDMotionGenerationResponse$ = exports.CreateSVDMotionGenerationResponse$outboundSchema = exports.CreateSVDMotionGenerationResponse$inboundSchema = exports.createSVDMotionGenerationResponseBodyFromJSON = exports.createSVDMotionGenerationResponseBodyToJSON = exports.CreateSVDMotionGenerationResponseBody$ = exports.CreateSVDMotionGenerationResponseBody$outboundSchema = exports.CreateSVDMotionGenerationResponseBody$inboundSchema = exports.motionSvdGenerationOutputFromJSON = exports.motionSvdGenerationOutputToJSON = exports.MotionSvdGenerationOutput$ = exports.MotionSvdGenerationOutput$outboundSchema = exports.MotionSvdGenerationOutput$inboundSchema = exports.createSVDMotionGenerationRequestBodyFromJSON = exports.createSVDMotionGenerationRequestBodyToJSON = exports.CreateSVDMotionGenerationRequestBody$ = exports.CreateSVDMotionGenerationRequestBody$outboundSchema = exports.CreateSVDMotionGenerationRequestBody$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.CreateSVDMotionGenerationRequestBody$inboundSchema = z.object({
    imageId: z.string(),
    isInitImage: z.nullable(z.boolean()).optional(),
    isPublic: z.nullable(z.boolean()).optional(),
    isVariation: z.nullable(z.boolean()).optional(),
    motionStrength: z.nullable(z.number().int()).optional(),
});
/** @internal */
exports.CreateSVDMotionGenerationRequestBody$outboundSchema = z.object({
    imageId: z.string(),
    isInitImage: z.nullable(z.boolean()).optional(),
    isPublic: z.nullable(z.boolean()).optional(),
    isVariation: z.nullable(z.boolean()).optional(),
    motionStrength: z.nullable(z.number().int()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateSVDMotionGenerationRequestBody$;
(function (CreateSVDMotionGenerationRequestBody$) {
    /** @deprecated use `CreateSVDMotionGenerationRequestBody$inboundSchema` instead. */
    CreateSVDMotionGenerationRequestBody$.inboundSchema = exports.CreateSVDMotionGenerationRequestBody$inboundSchema;
    /** @deprecated use `CreateSVDMotionGenerationRequestBody$outboundSchema` instead. */
    CreateSVDMotionGenerationRequestBody$.outboundSchema = exports.CreateSVDMotionGenerationRequestBody$outboundSchema;
})(CreateSVDMotionGenerationRequestBody$ || (exports.CreateSVDMotionGenerationRequestBody$ = CreateSVDMotionGenerationRequestBody$ = {}));
function createSVDMotionGenerationRequestBodyToJSON(createSVDMotionGenerationRequestBody) {
    return JSON.stringify(exports.CreateSVDMotionGenerationRequestBody$outboundSchema.parse(createSVDMotionGenerationRequestBody));
}
exports.createSVDMotionGenerationRequestBodyToJSON = createSVDMotionGenerationRequestBodyToJSON;
function createSVDMotionGenerationRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateSVDMotionGenerationRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSVDMotionGenerationRequestBody' from JSON`);
}
exports.createSVDMotionGenerationRequestBodyFromJSON = createSVDMotionGenerationRequestBodyFromJSON;
/** @internal */
exports.MotionSvdGenerationOutput$inboundSchema = z.object({
    apiCreditCost: z.nullable(z.number().int()).optional(),
    generationId: z.string().optional(),
});
/** @internal */
exports.MotionSvdGenerationOutput$outboundSchema = z.object({
    apiCreditCost: z.nullable(z.number().int()).optional(),
    generationId: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MotionSvdGenerationOutput$;
(function (MotionSvdGenerationOutput$) {
    /** @deprecated use `MotionSvdGenerationOutput$inboundSchema` instead. */
    MotionSvdGenerationOutput$.inboundSchema = exports.MotionSvdGenerationOutput$inboundSchema;
    /** @deprecated use `MotionSvdGenerationOutput$outboundSchema` instead. */
    MotionSvdGenerationOutput$.outboundSchema = exports.MotionSvdGenerationOutput$outboundSchema;
})(MotionSvdGenerationOutput$ || (exports.MotionSvdGenerationOutput$ = MotionSvdGenerationOutput$ = {}));
function motionSvdGenerationOutputToJSON(motionSvdGenerationOutput) {
    return JSON.stringify(exports.MotionSvdGenerationOutput$outboundSchema.parse(motionSvdGenerationOutput));
}
exports.motionSvdGenerationOutputToJSON = motionSvdGenerationOutputToJSON;
function motionSvdGenerationOutputFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MotionSvdGenerationOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MotionSvdGenerationOutput' from JSON`);
}
exports.motionSvdGenerationOutputFromJSON = motionSvdGenerationOutputFromJSON;
/** @internal */
exports.CreateSVDMotionGenerationResponseBody$inboundSchema = z.object({
    motionSvdGenerationJob: z.nullable(z.lazy(() => exports.MotionSvdGenerationOutput$inboundSchema)).optional(),
});
/** @internal */
exports.CreateSVDMotionGenerationResponseBody$outboundSchema = z.object({
    motionSvdGenerationJob: z.nullable(z.lazy(() => exports.MotionSvdGenerationOutput$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateSVDMotionGenerationResponseBody$;
(function (CreateSVDMotionGenerationResponseBody$) {
    /** @deprecated use `CreateSVDMotionGenerationResponseBody$inboundSchema` instead. */
    CreateSVDMotionGenerationResponseBody$.inboundSchema = exports.CreateSVDMotionGenerationResponseBody$inboundSchema;
    /** @deprecated use `CreateSVDMotionGenerationResponseBody$outboundSchema` instead. */
    CreateSVDMotionGenerationResponseBody$.outboundSchema = exports.CreateSVDMotionGenerationResponseBody$outboundSchema;
})(CreateSVDMotionGenerationResponseBody$ || (exports.CreateSVDMotionGenerationResponseBody$ = CreateSVDMotionGenerationResponseBody$ = {}));
function createSVDMotionGenerationResponseBodyToJSON(createSVDMotionGenerationResponseBody) {
    return JSON.stringify(exports.CreateSVDMotionGenerationResponseBody$outboundSchema.parse(createSVDMotionGenerationResponseBody));
}
exports.createSVDMotionGenerationResponseBodyToJSON = createSVDMotionGenerationResponseBodyToJSON;
function createSVDMotionGenerationResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateSVDMotionGenerationResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSVDMotionGenerationResponseBody' from JSON`);
}
exports.createSVDMotionGenerationResponseBodyFromJSON = createSVDMotionGenerationResponseBodyFromJSON;
/** @internal */
exports.CreateSVDMotionGenerationResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    object: z.lazy(() => exports.CreateSVDMotionGenerationResponseBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.CreateSVDMotionGenerationResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    object: z.lazy(() => exports.CreateSVDMotionGenerationResponseBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateSVDMotionGenerationResponse$;
(function (CreateSVDMotionGenerationResponse$) {
    /** @deprecated use `CreateSVDMotionGenerationResponse$inboundSchema` instead. */
    CreateSVDMotionGenerationResponse$.inboundSchema = exports.CreateSVDMotionGenerationResponse$inboundSchema;
    /** @deprecated use `CreateSVDMotionGenerationResponse$outboundSchema` instead. */
    CreateSVDMotionGenerationResponse$.outboundSchema = exports.CreateSVDMotionGenerationResponse$outboundSchema;
})(CreateSVDMotionGenerationResponse$ || (exports.CreateSVDMotionGenerationResponse$ = CreateSVDMotionGenerationResponse$ = {}));
function createSVDMotionGenerationResponseToJSON(createSVDMotionGenerationResponse) {
    return JSON.stringify(exports.CreateSVDMotionGenerationResponse$outboundSchema.parse(createSVDMotionGenerationResponse));
}
exports.createSVDMotionGenerationResponseToJSON = createSVDMotionGenerationResponseToJSON;
function createSVDMotionGenerationResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateSVDMotionGenerationResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSVDMotionGenerationResponse' from JSON`);
}
exports.createSVDMotionGenerationResponseFromJSON = createSVDMotionGenerationResponseFromJSON;
