// @leonardo-ai/sdk/sdk/models/operations/createvariationunzoom.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createVariationUnzoomResponseFromJSON = exports.createVariationUnzoomResponseToJSON = exports.CreateVariationUnzoomResponse$ = exports.CreateVariationUnzoomResponse$outboundSchema = exports.CreateVariationUnzoomResponse$inboundSchema = exports.createVariationUnzoomResponseBodyFromJSON = exports.createVariationUnzoomResponseBodyToJSON = exports.CreateVariationUnzoomResponseBody$ = exports.CreateVariationUnzoomResponseBody$outboundSchema = exports.CreateVariationUnzoomResponseBody$inboundSchema = exports.sdUnzoomOutputFromJSON = exports.sdUnzoomOutputToJSON = exports.SDUnzoomOutput$ = exports.SDUnzoomOutput$outboundSchema = exports.SDUnzoomOutput$inboundSchema = exports.createVariationUnzoomRequestBodyFromJSON = exports.createVariationUnzoomRequestBodyToJSON = exports.CreateVariationUnzoomRequestBody$ = exports.CreateVariationUnzoomRequestBody$outboundSchema = exports.CreateVariationUnzoomRequestBody$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.CreateVariationUnzoomRequestBody$inboundSchema = z.object({
    id: z.string().optional(),
    isVariation: z.nullable(z.boolean()).optional(),
});
/** @internal */
exports.CreateVariationUnzoomRequestBody$outboundSchema = z.object({
    id: z.string().optional(),
    isVariation: z.nullable(z.boolean()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateVariationUnzoomRequestBody$;
(function (CreateVariationUnzoomRequestBody$) {
    /** @deprecated use `CreateVariationUnzoomRequestBody$inboundSchema` instead. */
    CreateVariationUnzoomRequestBody$.inboundSchema = exports.CreateVariationUnzoomRequestBody$inboundSchema;
    /** @deprecated use `CreateVariationUnzoomRequestBody$outboundSchema` instead. */
    CreateVariationUnzoomRequestBody$.outboundSchema = exports.CreateVariationUnzoomRequestBody$outboundSchema;
})(CreateVariationUnzoomRequestBody$ || (exports.CreateVariationUnzoomRequestBody$ = CreateVariationUnzoomRequestBody$ = {}));
function createVariationUnzoomRequestBodyToJSON(createVariationUnzoomRequestBody) {
    return JSON.stringify(exports.CreateVariationUnzoomRequestBody$outboundSchema.parse(createVariationUnzoomRequestBody));
}
exports.createVariationUnzoomRequestBodyToJSON = createVariationUnzoomRequestBodyToJSON;
function createVariationUnzoomRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateVariationUnzoomRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateVariationUnzoomRequestBody' from JSON`);
}
exports.createVariationUnzoomRequestBodyFromJSON = createVariationUnzoomRequestBodyFromJSON;
/** @internal */
exports.SDUnzoomOutput$inboundSchema = z.object({
    apiCreditCost: z.nullable(z.number().int()).optional(),
    id: z.string().optional(),
});
/** @internal */
exports.SDUnzoomOutput$outboundSchema = z.object({
    apiCreditCost: z.nullable(z.number().int()).optional(),
    id: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SDUnzoomOutput$;
(function (SDUnzoomOutput$) {
    /** @deprecated use `SDUnzoomOutput$inboundSchema` instead. */
    SDUnzoomOutput$.inboundSchema = exports.SDUnzoomOutput$inboundSchema;
    /** @deprecated use `SDUnzoomOutput$outboundSchema` instead. */
    SDUnzoomOutput$.outboundSchema = exports.SDUnzoomOutput$outboundSchema;
})(SDUnzoomOutput$ || (exports.SDUnzoomOutput$ = SDUnzoomOutput$ = {}));
function sdUnzoomOutputToJSON(sdUnzoomOutput) {
    return JSON.stringify(exports.SDUnzoomOutput$outboundSchema.parse(sdUnzoomOutput));
}
exports.sdUnzoomOutputToJSON = sdUnzoomOutputToJSON;
function sdUnzoomOutputFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SDUnzoomOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SDUnzoomOutput' from JSON`);
}
exports.sdUnzoomOutputFromJSON = sdUnzoomOutputFromJSON;
/** @internal */
exports.CreateVariationUnzoomResponseBody$inboundSchema = z.object({
    sdUnzoomJob: z.nullable(z.lazy(() => exports.SDUnzoomOutput$inboundSchema))
        .optional(),
});
/** @internal */
exports.CreateVariationUnzoomResponseBody$outboundSchema = z.object({
    sdUnzoomJob: z.nullable(z.lazy(() => exports.SDUnzoomOutput$outboundSchema))
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateVariationUnzoomResponseBody$;
(function (CreateVariationUnzoomResponseBody$) {
    /** @deprecated use `CreateVariationUnzoomResponseBody$inboundSchema` instead. */
    CreateVariationUnzoomResponseBody$.inboundSchema = exports.CreateVariationUnzoomResponseBody$inboundSchema;
    /** @deprecated use `CreateVariationUnzoomResponseBody$outboundSchema` instead. */
    CreateVariationUnzoomResponseBody$.outboundSchema = exports.CreateVariationUnzoomResponseBody$outboundSchema;
})(CreateVariationUnzoomResponseBody$ || (exports.CreateVariationUnzoomResponseBody$ = CreateVariationUnzoomResponseBody$ = {}));
function createVariationUnzoomResponseBodyToJSON(createVariationUnzoomResponseBody) {
    return JSON.stringify(exports.CreateVariationUnzoomResponseBody$outboundSchema.parse(createVariationUnzoomResponseBody));
}
exports.createVariationUnzoomResponseBodyToJSON = createVariationUnzoomResponseBodyToJSON;
function createVariationUnzoomResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateVariationUnzoomResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateVariationUnzoomResponseBody' from JSON`);
}
exports.createVariationUnzoomResponseBodyFromJSON = createVariationUnzoomResponseBodyFromJSON;
/** @internal */
exports.CreateVariationUnzoomResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    object: z.lazy(() => exports.CreateVariationUnzoomResponseBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.CreateVariationUnzoomResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    object: z.lazy(() => exports.CreateVariationUnzoomResponseBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateVariationUnzoomResponse$;
(function (CreateVariationUnzoomResponse$) {
    /** @deprecated use `CreateVariationUnzoomResponse$inboundSchema` instead. */
    CreateVariationUnzoomResponse$.inboundSchema = exports.CreateVariationUnzoomResponse$inboundSchema;
    /** @deprecated use `CreateVariationUnzoomResponse$outboundSchema` instead. */
    CreateVariationUnzoomResponse$.outboundSchema = exports.CreateVariationUnzoomResponse$outboundSchema;
})(CreateVariationUnzoomResponse$ || (exports.CreateVariationUnzoomResponse$ = CreateVariationUnzoomResponse$ = {}));
function createVariationUnzoomResponseToJSON(createVariationUnzoomResponse) {
    return JSON.stringify(exports.CreateVariationUnzoomResponse$outboundSchema.parse(createVariationUnzoomResponse));
}
exports.createVariationUnzoomResponseToJSON = createVariationUnzoomResponseToJSON;
function createVariationUnzoomResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateVariationUnzoomResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateVariationUnzoomResponse' from JSON`);
}
exports.createVariationUnzoomResponseFromJSON = createVariationUnzoomResponseFromJSON;
