// @leonardo-ai/sdk/sdk/models/operations/getcustomelementsbyuserid.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCustomElementsByUserIdResponseFromJSON = exports.getCustomElementsByUserIdResponseToJSON = exports.GetCustomElementsByUserIdResponse$ = exports.GetCustomElementsByUserIdResponse$outboundSchema = exports.GetCustomElementsByUserIdResponse$inboundSchema = exports.getCustomElementsByUserIdResponseBodyFromJSON = exports.getCustomElementsByUserIdResponseBodyToJSON = exports.GetCustomElementsByUserIdResponseBody$ = exports.GetCustomElementsByUserIdResponseBody$outboundSchema = exports.GetCustomElementsByUserIdResponseBody$inboundSchema = exports.getCustomElementsByUserIdUserLorasFromJSON = exports.getCustomElementsByUserIdUserLorasToJSON = exports.GetCustomElementsByUserIdUserLoras$ = exports.GetCustomElementsByUserIdUserLoras$outboundSchema = exports.GetCustomElementsByUserIdUserLoras$inboundSchema = exports.getCustomElementsByUserIdRequestFromJSON = exports.getCustomElementsByUserIdRequestToJSON = exports.GetCustomElementsByUserIdRequest$ = exports.GetCustomElementsByUserIdRequest$outboundSchema = exports.GetCustomElementsByUserIdRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.GetCustomElementsByUserIdRequest$inboundSchema = z.object({
    userId: z.string(),
});
/** @internal */
exports.GetCustomElementsByUserIdRequest$outboundSchema = z.object({
    userId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCustomElementsByUserIdRequest$;
(function (GetCustomElementsByUserIdRequest$) {
    /** @deprecated use `GetCustomElementsByUserIdRequest$inboundSchema` instead. */
    GetCustomElementsByUserIdRequest$.inboundSchema = exports.GetCustomElementsByUserIdRequest$inboundSchema;
    /** @deprecated use `GetCustomElementsByUserIdRequest$outboundSchema` instead. */
    GetCustomElementsByUserIdRequest$.outboundSchema = exports.GetCustomElementsByUserIdRequest$outboundSchema;
})(GetCustomElementsByUserIdRequest$ || (exports.GetCustomElementsByUserIdRequest$ = GetCustomElementsByUserIdRequest$ = {}));
function getCustomElementsByUserIdRequestToJSON(getCustomElementsByUserIdRequest) {
    return JSON.stringify(exports.GetCustomElementsByUserIdRequest$outboundSchema.parse(getCustomElementsByUserIdRequest));
}
exports.getCustomElementsByUserIdRequestToJSON = getCustomElementsByUserIdRequestToJSON;
function getCustomElementsByUserIdRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCustomElementsByUserIdRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCustomElementsByUserIdRequest' from JSON`);
}
exports.getCustomElementsByUserIdRequestFromJSON = getCustomElementsByUserIdRequestFromJSON;
/** @internal */
exports.GetCustomElementsByUserIdUserLoras$inboundSchema = z.object({
    baseModel: z.string().optional(),
    createdAt: z.string().optional(),
    description: z.string().optional(),
    focus: z.string().optional(),
    id: z.number().int().optional(),
    instancePrompt: z.nullable(z.string()).optional(),
    learningRate: z.number().optional(),
    name: z.string().optional(),
    resolution: z.number().int().optional(),
    status: shared.JobStatus$inboundSchema.optional(),
    trainTextEncoder: z.boolean().optional(),
    trainingEpoch: z.number().int().optional(),
    updatedAt: z.string().optional(),
});
/** @internal */
exports.GetCustomElementsByUserIdUserLoras$outboundSchema = z.object({
    baseModel: z.string().optional(),
    createdAt: z.string().optional(),
    description: z.string().optional(),
    focus: z.string().optional(),
    id: z.number().int().optional(),
    instancePrompt: z.nullable(z.string()).optional(),
    learningRate: z.number().optional(),
    name: z.string().optional(),
    resolution: z.number().int().optional(),
    status: shared.JobStatus$outboundSchema.optional(),
    trainTextEncoder: z.boolean().optional(),
    trainingEpoch: z.number().int().optional(),
    updatedAt: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCustomElementsByUserIdUserLoras$;
(function (GetCustomElementsByUserIdUserLoras$) {
    /** @deprecated use `GetCustomElementsByUserIdUserLoras$inboundSchema` instead. */
    GetCustomElementsByUserIdUserLoras$.inboundSchema = exports.GetCustomElementsByUserIdUserLoras$inboundSchema;
    /** @deprecated use `GetCustomElementsByUserIdUserLoras$outboundSchema` instead. */
    GetCustomElementsByUserIdUserLoras$.outboundSchema = exports.GetCustomElementsByUserIdUserLoras$outboundSchema;
})(GetCustomElementsByUserIdUserLoras$ || (exports.GetCustomElementsByUserIdUserLoras$ = GetCustomElementsByUserIdUserLoras$ = {}));
function getCustomElementsByUserIdUserLorasToJSON(getCustomElementsByUserIdUserLoras) {
    return JSON.stringify(exports.GetCustomElementsByUserIdUserLoras$outboundSchema.parse(getCustomElementsByUserIdUserLoras));
}
exports.getCustomElementsByUserIdUserLorasToJSON = getCustomElementsByUserIdUserLorasToJSON;
function getCustomElementsByUserIdUserLorasFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCustomElementsByUserIdUserLoras$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCustomElementsByUserIdUserLoras' from JSON`);
}
exports.getCustomElementsByUserIdUserLorasFromJSON = getCustomElementsByUserIdUserLorasFromJSON;
/** @internal */
exports.GetCustomElementsByUserIdResponseBody$inboundSchema = z.object({
    user_loras: z.nullable(z.array(z.lazy(() => exports.GetCustomElementsByUserIdUserLoras$inboundSchema))).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "user_loras": "userLoras",
    });
});
/** @internal */
exports.GetCustomElementsByUserIdResponseBody$outboundSchema = z.object({
    userLoras: z.nullable(z.array(z.lazy(() => exports.GetCustomElementsByUserIdUserLoras$outboundSchema))).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        userLoras: "user_loras",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCustomElementsByUserIdResponseBody$;
(function (GetCustomElementsByUserIdResponseBody$) {
    /** @deprecated use `GetCustomElementsByUserIdResponseBody$inboundSchema` instead. */
    GetCustomElementsByUserIdResponseBody$.inboundSchema = exports.GetCustomElementsByUserIdResponseBody$inboundSchema;
    /** @deprecated use `GetCustomElementsByUserIdResponseBody$outboundSchema` instead. */
    GetCustomElementsByUserIdResponseBody$.outboundSchema = exports.GetCustomElementsByUserIdResponseBody$outboundSchema;
})(GetCustomElementsByUserIdResponseBody$ || (exports.GetCustomElementsByUserIdResponseBody$ = GetCustomElementsByUserIdResponseBody$ = {}));
function getCustomElementsByUserIdResponseBodyToJSON(getCustomElementsByUserIdResponseBody) {
    return JSON.stringify(exports.GetCustomElementsByUserIdResponseBody$outboundSchema.parse(getCustomElementsByUserIdResponseBody));
}
exports.getCustomElementsByUserIdResponseBodyToJSON = getCustomElementsByUserIdResponseBodyToJSON;
function getCustomElementsByUserIdResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCustomElementsByUserIdResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCustomElementsByUserIdResponseBody' from JSON`);
}
exports.getCustomElementsByUserIdResponseBodyFromJSON = getCustomElementsByUserIdResponseBodyFromJSON;
/** @internal */
exports.GetCustomElementsByUserIdResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    object: z.lazy(() => exports.GetCustomElementsByUserIdResponseBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.GetCustomElementsByUserIdResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    object: z.lazy(() => exports.GetCustomElementsByUserIdResponseBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCustomElementsByUserIdResponse$;
(function (GetCustomElementsByUserIdResponse$) {
    /** @deprecated use `GetCustomElementsByUserIdResponse$inboundSchema` instead. */
    GetCustomElementsByUserIdResponse$.inboundSchema = exports.GetCustomElementsByUserIdResponse$inboundSchema;
    /** @deprecated use `GetCustomElementsByUserIdResponse$outboundSchema` instead. */
    GetCustomElementsByUserIdResponse$.outboundSchema = exports.GetCustomElementsByUserIdResponse$outboundSchema;
})(GetCustomElementsByUserIdResponse$ || (exports.GetCustomElementsByUserIdResponse$ = GetCustomElementsByUserIdResponse$ = {}));
function getCustomElementsByUserIdResponseToJSON(getCustomElementsByUserIdResponse) {
    return JSON.stringify(exports.GetCustomElementsByUserIdResponse$outboundSchema.parse(getCustomElementsByUserIdResponse));
}
exports.getCustomElementsByUserIdResponseToJSON = getCustomElementsByUserIdResponseToJSON;
function getCustomElementsByUserIdResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCustomElementsByUserIdResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCustomElementsByUserIdResponse' from JSON`);
}
exports.getCustomElementsByUserIdResponseFromJSON = getCustomElementsByUserIdResponseFromJSON;
