// @leonardo-ai/sdk/sdk/models/operations/pricingcalculator.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.pricingCalculatorRequestBodyFromJSON = exports.pricingCalculatorRequestBodyToJSON = exports.PricingCalculatorRequestBody$ = exports.PricingCalculatorRequestBody$outboundSchema = exports.PricingCalculatorRequestBody$inboundSchema = exports.objectFromJSON = exports.objectToJSON = exports.ObjectT$ = exports.ObjectT$outboundSchema = exports.ObjectT$inboundSchema = exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObjectFromJSON = exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObjectToJSON = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$ = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$outboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$inboundSchema = exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObjectFromJSON = exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObjectToJSON = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$ = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$outboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$inboundSchema = exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObjectFromJSON = exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObjectToJSON = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$ = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$outboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$inboundSchema = exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObjectFromJSON = exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObjectToJSON = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$ = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$outboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$inboundSchema = exports.pricingCalculatorPricingCalculatorRequestRequestBodyObjectFromJSON = exports.pricingCalculatorPricingCalculatorRequestRequestBodyObjectToJSON = exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$ = exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$outboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$inboundSchema = exports.pricingCalculatorPricingCalculatorRequestObjectFromJSON = exports.pricingCalculatorPricingCalculatorRequestObjectToJSON = exports.PricingCalculatorPricingCalculatorRequestObject$ = exports.PricingCalculatorPricingCalculatorRequestObject$outboundSchema = exports.PricingCalculatorPricingCalculatorRequestObject$inboundSchema = exports.pricingCalculatorPricingCalculatorObjectFromJSON = exports.pricingCalculatorPricingCalculatorObjectToJSON = exports.PricingCalculatorPricingCalculatorObject$ = exports.PricingCalculatorPricingCalculatorObject$outboundSchema = exports.PricingCalculatorPricingCalculatorObject$inboundSchema = exports.pricingCalculatorObjectFromJSON = exports.pricingCalculatorObjectToJSON = exports.PricingCalculatorObject$ = exports.PricingCalculatorObject$outboundSchema = exports.PricingCalculatorObject$inboundSchema = void 0;
exports.pricingCalculatorResponseFromJSON = exports.pricingCalculatorResponseToJSON = exports.PricingCalculatorResponse$ = exports.PricingCalculatorResponse$outboundSchema = exports.PricingCalculatorResponse$inboundSchema = exports.pricingCalculatorResponseBodyFromJSON = exports.pricingCalculatorResponseBodyToJSON = exports.PricingCalculatorResponseBody$ = exports.PricingCalculatorResponseBody$outboundSchema = exports.PricingCalculatorResponseBody$inboundSchema = exports.calculateProductionApiServiceCostFromJSON = exports.calculateProductionApiServiceCostToJSON = exports.CalculateProductionApiServiceCost$ = exports.CalculateProductionApiServiceCost$outboundSchema = exports.CalculateProductionApiServiceCost$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.PricingCalculatorObject$inboundSchema = z.object({
    imageHeight: z.number().int().optional(),
    imageWidth: z.number().int().optional(),
    numImages: z.number().int().optional(),
});
/** @internal */
exports.PricingCalculatorObject$outboundSchema = z.object({
    imageHeight: z.number().int().optional(),
    imageWidth: z.number().int().optional(),
    numImages: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorObject$;
(function (PricingCalculatorObject$) {
    /** @deprecated use `PricingCalculatorObject$inboundSchema` instead. */
    PricingCalculatorObject$.inboundSchema = exports.PricingCalculatorObject$inboundSchema;
    /** @deprecated use `PricingCalculatorObject$outboundSchema` instead. */
    PricingCalculatorObject$.outboundSchema = exports.PricingCalculatorObject$outboundSchema;
})(PricingCalculatorObject$ || (exports.PricingCalculatorObject$ = PricingCalculatorObject$ = {}));
function pricingCalculatorObjectToJSON(pricingCalculatorObject) {
    return JSON.stringify(exports.PricingCalculatorObject$outboundSchema.parse(pricingCalculatorObject));
}
exports.pricingCalculatorObjectToJSON = pricingCalculatorObjectToJSON;
function pricingCalculatorObjectFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PricingCalculatorObject$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PricingCalculatorObject' from JSON`);
}
exports.pricingCalculatorObjectFromJSON = pricingCalculatorObjectFromJSON;
/** @internal */
exports.PricingCalculatorPricingCalculatorObject$inboundSchema = z.object({
    alchemyMode: z.boolean().optional(),
    controlnetsCost: z.nullable(z.number().int()).optional(),
    highResolution: z.boolean().optional(),
    imageHeight: z.number().int().optional(),
    imageWidth: z.number().int().optional(),
    inferenceSteps: z.number().int().optional(),
    isModelCustom: z.nullable(z.boolean()).optional(),
    isPhoenix: z.nullable(z.boolean()).optional(),
    isSDXL: z.nullable(z.boolean()).optional(),
    isSDXLLightning: z.nullable(z.boolean()).optional(),
    loraCount: z.nullable(z.number().int()).optional(),
    numImages: z.number().int().optional(),
    photoRealMode: z.nullable(z.boolean()).optional(),
    photoRealStrength: z.nullable(z.number()).optional(),
    photoRealVersion: z.nullable(z.string()).optional(),
    promptMagic: z.nullable(z.boolean()).optional(),
    promptMagicStrength: z.nullable(z.number()).optional(),
    promptMagicVersion: z.nullable(z.string()).optional(),
    ultra: z.nullable(z.boolean()).optional(),
});
/** @internal */
exports.PricingCalculatorPricingCalculatorObject$outboundSchema = z.object({
    alchemyMode: z.boolean().optional(),
    controlnetsCost: z.nullable(z.number().int()).optional(),
    highResolution: z.boolean().optional(),
    imageHeight: z.number().int().optional(),
    imageWidth: z.number().int().optional(),
    inferenceSteps: z.number().int().optional(),
    isModelCustom: z.nullable(z.boolean()).optional(),
    isPhoenix: z.nullable(z.boolean()).optional(),
    isSDXL: z.nullable(z.boolean()).optional(),
    isSDXLLightning: z.nullable(z.boolean()).optional(),
    loraCount: z.nullable(z.number().int()).optional(),
    numImages: z.number().int().optional(),
    photoRealMode: z.nullable(z.boolean()).optional(),
    photoRealStrength: z.nullable(z.number()).optional(),
    photoRealVersion: z.nullable(z.string()).optional(),
    promptMagic: z.nullable(z.boolean()).optional(),
    promptMagicStrength: z.nullable(z.number()).optional(),
    promptMagicVersion: z.nullable(z.string()).optional(),
    ultra: z.nullable(z.boolean()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorPricingCalculatorObject$;
(function (PricingCalculatorPricingCalculatorObject$) {
    /** @deprecated use `PricingCalculatorPricingCalculatorObject$inboundSchema` instead. */
    PricingCalculatorPricingCalculatorObject$.inboundSchema = exports.PricingCalculatorPricingCalculatorObject$inboundSchema;
    /** @deprecated use `PricingCalculatorPricingCalculatorObject$outboundSchema` instead. */
    PricingCalculatorPricingCalculatorObject$.outboundSchema = exports.PricingCalculatorPricingCalculatorObject$outboundSchema;
})(PricingCalculatorPricingCalculatorObject$ || (exports.PricingCalculatorPricingCalculatorObject$ = PricingCalculatorPricingCalculatorObject$ = {}));
function pricingCalculatorPricingCalculatorObjectToJSON(pricingCalculatorPricingCalculatorObject) {
    return JSON.stringify(exports.PricingCalculatorPricingCalculatorObject$outboundSchema.parse(pricingCalculatorPricingCalculatorObject));
}
exports.pricingCalculatorPricingCalculatorObjectToJSON = pricingCalculatorPricingCalculatorObjectToJSON;
function pricingCalculatorPricingCalculatorObjectFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PricingCalculatorPricingCalculatorObject$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PricingCalculatorPricingCalculatorObject' from JSON`);
}
exports.pricingCalculatorPricingCalculatorObjectFromJSON = pricingCalculatorPricingCalculatorObjectFromJSON;
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestObject$inboundSchema = z.object({
    height: z.nullable(z.number().int()).optional(),
    instantRefine: z.nullable(z.boolean()).optional(),
    refine: z.nullable(z.boolean()).optional(),
    width: z.nullable(z.number().int()).optional(),
});
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestObject$outboundSchema = z.object({
    height: z.nullable(z.number().int()).optional(),
    instantRefine: z.nullable(z.boolean()).optional(),
    refine: z.nullable(z.boolean()).optional(),
    width: z.nullable(z.number().int()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorPricingCalculatorRequestObject$;
(function (PricingCalculatorPricingCalculatorRequestObject$) {
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestObject$inboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestObject$.inboundSchema = exports.PricingCalculatorPricingCalculatorRequestObject$inboundSchema;
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestObject$outboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestObject$.outboundSchema = exports.PricingCalculatorPricingCalculatorRequestObject$outboundSchema;
})(PricingCalculatorPricingCalculatorRequestObject$ || (exports.PricingCalculatorPricingCalculatorRequestObject$ = PricingCalculatorPricingCalculatorRequestObject$ = {}));
function pricingCalculatorPricingCalculatorRequestObjectToJSON(pricingCalculatorPricingCalculatorRequestObject) {
    return JSON.stringify(exports.PricingCalculatorPricingCalculatorRequestObject$outboundSchema.parse(pricingCalculatorPricingCalculatorRequestObject));
}
exports.pricingCalculatorPricingCalculatorRequestObjectToJSON = pricingCalculatorPricingCalculatorRequestObjectToJSON;
function pricingCalculatorPricingCalculatorRequestObjectFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PricingCalculatorPricingCalculatorRequestObject$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PricingCalculatorPricingCalculatorRequestObject' from JSON`);
}
exports.pricingCalculatorPricingCalculatorRequestObjectFromJSON = pricingCalculatorPricingCalculatorRequestObjectFromJSON;
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$inboundSchema = z.object({
    resolution: z.number().int().optional(),
});
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$outboundSchema = z.object({
    resolution: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorPricingCalculatorRequestRequestBodyObject$;
(function (PricingCalculatorPricingCalculatorRequestRequestBodyObject$) {
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestRequestBodyObject$inboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestRequestBodyObject$.inboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$inboundSchema;
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestRequestBodyObject$outboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestRequestBodyObject$.outboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$outboundSchema;
})(PricingCalculatorPricingCalculatorRequestRequestBodyObject$ || (exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$ = PricingCalculatorPricingCalculatorRequestRequestBodyObject$ = {}));
function pricingCalculatorPricingCalculatorRequestRequestBodyObjectToJSON(pricingCalculatorPricingCalculatorRequestRequestBodyObject) {
    return JSON.stringify(exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$outboundSchema
        .parse(pricingCalculatorPricingCalculatorRequestRequestBodyObject));
}
exports.pricingCalculatorPricingCalculatorRequestRequestBodyObjectToJSON = pricingCalculatorPricingCalculatorRequestRequestBodyObjectToJSON;
function pricingCalculatorPricingCalculatorRequestRequestBodyObjectFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'PricingCalculatorPricingCalculatorRequestRequestBodyObject' from JSON`);
}
exports.pricingCalculatorPricingCalculatorRequestRequestBodyObjectFromJSON = pricingCalculatorPricingCalculatorRequestRequestBodyObjectFromJSON;
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$inboundSchema = z.object({
    durationSeconds: z.number().int().optional(),
});
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$outboundSchema = z.object({
    durationSeconds: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$;
(function (PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$) {
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$inboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$.inboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$inboundSchema;
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$outboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$.outboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$outboundSchema;
})(PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$ || (exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$ = PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$ = {}));
function pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObjectToJSON(pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject) {
    return JSON.stringify(exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$outboundSchema
        .parse(pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject));
}
exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObjectToJSON = pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObjectToJSON;
function pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObjectFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject' from JSON`);
}
exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObjectFromJSON = pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObjectFromJSON;
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$inboundSchema = z.object({
    preview: z.boolean().optional(),
});
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$outboundSchema = z.object({
    preview: z.boolean().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$;
(function (PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$) {
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$inboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$.inboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$inboundSchema;
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$outboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$.outboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$outboundSchema;
})(PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$ || (exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$ = PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$ = {}));
function pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObjectToJSON(pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject) {
    return JSON.stringify(exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$outboundSchema
        .parse(pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject));
}
exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObjectToJSON = pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObjectToJSON;
function pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObjectFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject' from JSON`);
}
exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObjectFromJSON = pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObjectFromJSON;
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$inboundSchema = z.object({
    megapixel: z.number().int().optional(),
});
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$outboundSchema = z.object({
    megapixel: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$;
(function (PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$) {
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$inboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$.inboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$inboundSchema;
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$outboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$.outboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$outboundSchema;
})(PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$ || (exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$ = PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$ = {}));
function pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObjectToJSON(pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject) {
    return JSON.stringify(exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$outboundSchema
        .parse(pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject));
}
exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObjectToJSON = pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObjectToJSON;
function pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObjectFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject' from JSON`);
}
exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObjectFromJSON = pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObjectFromJSON;
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$inboundSchema = z.object({
    inputHeight: z.number().int().optional(),
    inputWidth: z.number().int().optional(),
    upscaleMultiplier: z.number().default(1.5),
});
/** @internal */
exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$outboundSchema = z.object({
    inputHeight: z.number().int().optional(),
    inputWidth: z.number().int().optional(),
    upscaleMultiplier: z.number().default(1.5),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$;
(function (PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$) {
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$inboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$.inboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$inboundSchema;
    /** @deprecated use `PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$outboundSchema` instead. */
    PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$.outboundSchema = exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$outboundSchema;
})(PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$ || (exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$ = PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$ = {}));
function pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObjectToJSON(pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject) {
    return JSON.stringify(exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$outboundSchema
        .parse(pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject));
}
exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObjectToJSON = pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObjectToJSON;
function pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObjectFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject' from JSON`);
}
exports.pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObjectFromJSON = pricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObjectFromJSON;
/** @internal */
exports.ObjectT$inboundSchema = z.object({
    FANTASY_AVATAR_GENERATION: z.nullable(z.lazy(() => exports.PricingCalculatorObject$inboundSchema)).optional(),
    IMAGE_GENERATION: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorObject$inboundSchema)).optional(),
    LCM_GENERATION: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestObject$inboundSchema)).optional(),
    MODEL_TRAINING: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$inboundSchema)).optional(),
    MOTION_GENERATION: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$inboundSchema)).optional(),
    TEXTURE_GENERATION: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$inboundSchema)).optional(),
    UNIVERSAL_UPSCALER: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$inboundSchema)).optional(),
    UNIVERSAL_UPSCALER_ULTRA: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "FANTASY_AVATAR_GENERATION": "fantasyAvatarGeneration",
        "IMAGE_GENERATION": "imageGeneration",
        "LCM_GENERATION": "lcmGeneration",
        "MODEL_TRAINING": "modelTraining",
        "MOTION_GENERATION": "motionGeneration",
        "TEXTURE_GENERATION": "textureGeneration",
        "UNIVERSAL_UPSCALER": "universalUpscaler",
        "UNIVERSAL_UPSCALER_ULTRA": "universalUpscalerUltra",
    });
});
/** @internal */
exports.ObjectT$outboundSchema = z.object({
    fantasyAvatarGeneration: z.nullable(z.lazy(() => exports.PricingCalculatorObject$outboundSchema)).optional(),
    imageGeneration: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorObject$outboundSchema)).optional(),
    lcmGeneration: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestObject$outboundSchema)).optional(),
    modelTraining: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestRequestBodyObject$outboundSchema)).optional(),
    motionGeneration: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsObject$outboundSchema)).optional(),
    textureGeneration: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsTEXTUREGENERATIONObject$outboundSchema)).optional(),
    universalUpscaler: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERObject$outboundSchema)).optional(),
    universalUpscalerUltra: z.nullable(z.lazy(() => exports.PricingCalculatorPricingCalculatorRequestRequestBodyServiceParamsUNIVERSALUPSCALERULTRAObject$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        fantasyAvatarGeneration: "FANTASY_AVATAR_GENERATION",
        imageGeneration: "IMAGE_GENERATION",
        lcmGeneration: "LCM_GENERATION",
        modelTraining: "MODEL_TRAINING",
        motionGeneration: "MOTION_GENERATION",
        textureGeneration: "TEXTURE_GENERATION",
        universalUpscaler: "UNIVERSAL_UPSCALER",
        universalUpscalerUltra: "UNIVERSAL_UPSCALER_ULTRA",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ObjectT$;
(function (ObjectT$) {
    /** @deprecated use `ObjectT$inboundSchema` instead. */
    ObjectT$.inboundSchema = exports.ObjectT$inboundSchema;
    /** @deprecated use `ObjectT$outboundSchema` instead. */
    ObjectT$.outboundSchema = exports.ObjectT$outboundSchema;
})(ObjectT$ || (exports.ObjectT$ = ObjectT$ = {}));
function objectToJSON(objectT) {
    return JSON.stringify(exports.ObjectT$outboundSchema.parse(objectT));
}
exports.objectToJSON = objectToJSON;
function objectFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ObjectT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObjectT' from JSON`);
}
exports.objectFromJSON = objectFromJSON;
/** @internal */
exports.PricingCalculatorRequestBody$inboundSchema = z.object({
    service: shared.PricingCalculatorServices$inboundSchema.optional(),
    serviceParams: z.lazy(() => exports.ObjectT$inboundSchema).optional(),
});
/** @internal */
exports.PricingCalculatorRequestBody$outboundSchema = z.object({
    service: shared.PricingCalculatorServices$outboundSchema.optional(),
    serviceParams: z.lazy(() => exports.ObjectT$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorRequestBody$;
(function (PricingCalculatorRequestBody$) {
    /** @deprecated use `PricingCalculatorRequestBody$inboundSchema` instead. */
    PricingCalculatorRequestBody$.inboundSchema = exports.PricingCalculatorRequestBody$inboundSchema;
    /** @deprecated use `PricingCalculatorRequestBody$outboundSchema` instead. */
    PricingCalculatorRequestBody$.outboundSchema = exports.PricingCalculatorRequestBody$outboundSchema;
})(PricingCalculatorRequestBody$ || (exports.PricingCalculatorRequestBody$ = PricingCalculatorRequestBody$ = {}));
function pricingCalculatorRequestBodyToJSON(pricingCalculatorRequestBody) {
    return JSON.stringify(exports.PricingCalculatorRequestBody$outboundSchema.parse(pricingCalculatorRequestBody));
}
exports.pricingCalculatorRequestBodyToJSON = pricingCalculatorRequestBodyToJSON;
function pricingCalculatorRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PricingCalculatorRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PricingCalculatorRequestBody' from JSON`);
}
exports.pricingCalculatorRequestBodyFromJSON = pricingCalculatorRequestBodyFromJSON;
/** @internal */
exports.CalculateProductionApiServiceCost$inboundSchema = z.object({
    cost: z.number().int().optional(),
});
/** @internal */
exports.CalculateProductionApiServiceCost$outboundSchema = z.object({
    cost: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CalculateProductionApiServiceCost$;
(function (CalculateProductionApiServiceCost$) {
    /** @deprecated use `CalculateProductionApiServiceCost$inboundSchema` instead. */
    CalculateProductionApiServiceCost$.inboundSchema = exports.CalculateProductionApiServiceCost$inboundSchema;
    /** @deprecated use `CalculateProductionApiServiceCost$outboundSchema` instead. */
    CalculateProductionApiServiceCost$.outboundSchema = exports.CalculateProductionApiServiceCost$outboundSchema;
})(CalculateProductionApiServiceCost$ || (exports.CalculateProductionApiServiceCost$ = CalculateProductionApiServiceCost$ = {}));
function calculateProductionApiServiceCostToJSON(calculateProductionApiServiceCost) {
    return JSON.stringify(exports.CalculateProductionApiServiceCost$outboundSchema.parse(calculateProductionApiServiceCost));
}
exports.calculateProductionApiServiceCostToJSON = calculateProductionApiServiceCostToJSON;
function calculateProductionApiServiceCostFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CalculateProductionApiServiceCost$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CalculateProductionApiServiceCost' from JSON`);
}
exports.calculateProductionApiServiceCostFromJSON = calculateProductionApiServiceCostFromJSON;
/** @internal */
exports.PricingCalculatorResponseBody$inboundSchema = z.object({
    calculateProductionApiServiceCost: z.nullable(z.lazy(() => exports.CalculateProductionApiServiceCost$inboundSchema)).optional(),
});
/** @internal */
exports.PricingCalculatorResponseBody$outboundSchema = z.object({
    calculateProductionApiServiceCost: z.nullable(z.lazy(() => exports.CalculateProductionApiServiceCost$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorResponseBody$;
(function (PricingCalculatorResponseBody$) {
    /** @deprecated use `PricingCalculatorResponseBody$inboundSchema` instead. */
    PricingCalculatorResponseBody$.inboundSchema = exports.PricingCalculatorResponseBody$inboundSchema;
    /** @deprecated use `PricingCalculatorResponseBody$outboundSchema` instead. */
    PricingCalculatorResponseBody$.outboundSchema = exports.PricingCalculatorResponseBody$outboundSchema;
})(PricingCalculatorResponseBody$ || (exports.PricingCalculatorResponseBody$ = PricingCalculatorResponseBody$ = {}));
function pricingCalculatorResponseBodyToJSON(pricingCalculatorResponseBody) {
    return JSON.stringify(exports.PricingCalculatorResponseBody$outboundSchema.parse(pricingCalculatorResponseBody));
}
exports.pricingCalculatorResponseBodyToJSON = pricingCalculatorResponseBodyToJSON;
function pricingCalculatorResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PricingCalculatorResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PricingCalculatorResponseBody' from JSON`);
}
exports.pricingCalculatorResponseBodyFromJSON = pricingCalculatorResponseBodyFromJSON;
/** @internal */
exports.PricingCalculatorResponse$inboundSchema = z.object({
    ContentType: z.string(),
    StatusCode: z.number().int(),
    RawResponse: z.instanceof(Response),
    object: z.lazy(() => exports.PricingCalculatorResponseBody$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ContentType": "contentType",
        "StatusCode": "statusCode",
        "RawResponse": "rawResponse",
    });
});
/** @internal */
exports.PricingCalculatorResponse$outboundSchema = z.object({
    contentType: z.string(),
    statusCode: z.number().int(),
    rawResponse: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    object: z.lazy(() => exports.PricingCalculatorResponseBody$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contentType: "ContentType",
        statusCode: "StatusCode",
        rawResponse: "RawResponse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PricingCalculatorResponse$;
(function (PricingCalculatorResponse$) {
    /** @deprecated use `PricingCalculatorResponse$inboundSchema` instead. */
    PricingCalculatorResponse$.inboundSchema = exports.PricingCalculatorResponse$inboundSchema;
    /** @deprecated use `PricingCalculatorResponse$outboundSchema` instead. */
    PricingCalculatorResponse$.outboundSchema = exports.PricingCalculatorResponse$outboundSchema;
})(PricingCalculatorResponse$ || (exports.PricingCalculatorResponse$ = PricingCalculatorResponse$ = {}));
function pricingCalculatorResponseToJSON(pricingCalculatorResponse) {
    return JSON.stringify(exports.PricingCalculatorResponse$outboundSchema.parse(pricingCalculatorResponse));
}
exports.pricingCalculatorResponseToJSON = pricingCalculatorResponseToJSON;
function pricingCalculatorResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PricingCalculatorResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PricingCalculatorResponse' from JSON`);
}
exports.pricingCalculatorResponseFromJSON = pricingCalculatorResponseFromJSON;
