// @leonardo-ai/sdk/lib/config.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SDK_METADATA = exports.serverURLFromOptions = exports.ServerList = void 0;
const url_js_1 = require("./url.js");
/**
 * Contains the list of servers available to the SDK
 */
exports.ServerList = [
    /**
     * Leonardo.Ai API server
     */
    "https://cloud.leonardo.ai/api/rest/v1",
];
function serverURLFromOptions(options) {
    let serverURL = options.serverURL;
    const params = {};
    if (!serverURL) {
        const serverIdx = options.serverIdx ?? 0;
        if (serverIdx < 0 || serverIdx >= exports.ServerList.length) {
            throw new Error(`Invalid server index ${serverIdx}`);
        }
        serverURL = exports.ServerList[serverIdx] || "";
    }
    const u = (0, url_js_1.pathToFunc)(serverURL)(params);
    return new URL(u);
}
exports.serverURLFromOptions = serverURLFromOptions;
exports.SDK_METADATA = {
    language: "typescript",
    openapiDocVersion: "v1.0.0",
    sdkVersion: "4.20.0",
    genVersion: "2.506.0",
    userAgent: "speakeasy-sdk/typescript 4.20.0 2.506.0 v1.0.0 @leonardo-ai/sdk",
};
